src/plugins/zoom.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * Zoom plugin
+ *
+ * zoom player when mouse wheel
+ *
+ * @author HoodyHuo (https://github.com/HoodyHuo)
+ *
+ * @example
+ * // ... initialising wavesurfer with the plugin
+ * var wavesurfer = WaveSurfer.create({
+ *   // wavesurfer options ...
+ *   plugins: [
+ *     ZoomPlugin.create({
+ *       // plugin options ...
+ *     })
+ *   ]
+ * });
+ */
+
+import { BasePlugin, BasePluginEvents } from '../base-plugin.js'
+
+export type ZoomPluginOptions = {
+  scale?: number // every wheel step zoom  0.1 means 10%
+}
+const defaultOptions = {
+  scale: 0.2,
+}
+
+export type ZoomPluginEvents = BasePluginEvents
+
+class ZoomPlugin extends BasePlugin<ZoomPluginEvents, ZoomPluginOptions> {
+  protected options: ZoomPluginOptions & typeof defaultOptions
+  private wrapper: HTMLElement | undefined = undefined
+  private container: HTMLElement | null = null
+
+  constructor(options?: ZoomPluginOptions) {
+    super(options || {})
+    this.options = Object.assign({}, defaultOptions, options)
+  }
+
+  public static create(options?: ZoomPluginOptions) {
+    return new ZoomPlugin(options)
+  }
+
+  onInit() {
+    this.wrapper = this.wavesurfer?.getWrapper()
+    if (!this.wrapper) {
+      return
+    }
+    if (this.wavesurfer?.options.container) {
+      const containerParam = this.wavesurfer?.options.container
+      if (typeof containerParam === 'string') {
+        this.container = document.querySelector(containerParam) as HTMLElement
+      } else if (containerParam instanceof HTMLElement) {
+        this.container = containerParam
+      } else {
+        this.container = this.wavesurfer?.getWrapper().parentElement as HTMLElement
+      }
+    }
```

Comment:
```suggestion
    this.container = this.wrapper.parentElement as HTMLElement
```

The container width is not always the width of the resulting waveform, so it's better to use the parent in all cases.